### PR TITLE
SOLR-16231: Fix links to ref-guide pages

### PR DESCRIPTION
**Pull request by HoustonPutman**
https://issues.apache.org/jira/browse/SOLR-16231

As of the 9.0.0 release, our ref-guide page links have changed considerably. We have created redirects for most of these, but still the codebase should use the correct link structure.

Note: I have found some redirects that don't work, so I will jot them down and we can try to fix them separately.

Still many to do.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ result faceting and highlighting.
 ## Online Documentation
 
 This README file only contains basic setup instructions.  For more
-comprehensive documentation, visit <https://solr.apache.org/guide/>
+comprehensive documentation, visit <https://solr.apache.org/guide/solr>
```
Comment (epugh) [inline README.md:31]: I tested, and `https://solr.apache.org/guide/` works, `https://solr.apache.org/guide/solr` doesn't....

Comment (epugh) [reply README.md:31]: However, https://solr.apache.org/guide/solr/latest/configuration-guide/solr-plugins.html works, and https://solr.apache.org/guide/latest/configuration-guide/solr-plugins.html doesn't, so maybe this is okay?

Comment (epugh) [reply README.md:31]: Nevermind, I see you made a change later on to make the urls have /solr/ ;-)

**Review by epugh — APPROVED** [review]
Thanks for chasing this one more time!  Links have been kind of a whack-a-mole problem between 8 and 9 ;-)

File: README.md (hunk 1 above)
Comment (HoustonPutman) [reply README.md:31]: Yeah antora comes with a default htaccess rule to redirect `/guide` to `/guide/solr/latest/index.html`. For some reason it doesn't also redirect `/guide/solr` to `/guide/solr/latest/index.html`. So I'm going to add that to solr site as well.

**Pull request merged**